lib/tool_shed/galaxy_install/tool_dependencies/recipe/step_handler.py

```diff
@@ -852,7 +852,11 @@ def move_directory_files( self, current_dir, source_dir, destination_dir ):
             os.makedirs(destination_directory)
         for dir_entry in os.listdir(source_directory):
             source_entry = os.path.join(source_directory, dir_entry)
-            if os.path.islink(source_entry):
+            if os.path.isdir(source_entry):
+                destination_subdir = os.path.join(destination_directory, dir_entry)
+                if not os.path.exists(destination_subdir):
+                    os.makedirs(destination_subdir)
+            elif os.path.islink(source_entry):
```

Comment:
I see, so the [commit](https://github.com/galaxyproject/galaxy/commit/d38936bd8d9b3612dc0718859067e299651e6c86) that may have broken the subdirectory handling was in response to https://github.com/galaxyproject/galaxy/issues/1947.
So I think that if we revert to the original code, but changing
```
        for source_file, destination_file in symlinks:
            shutil.move( source_file, destination_file )
```
to 
```
        for source_file, destination_file in symlinks:
            os.symlink(os.readlink(source_file), destination_file)
            shutil.rm( source_file )
```
We should have both #1947 fixed and the subdirectory problem solved, right?